src/GraphUtils.ts

```diff
@@ -426,16 +430,31 @@ export function findTurnPts(pts: Point[], mode: string, isClosed: boolean = fals
         if (pts[i].y > pts[mod(i-1, pts.length)].y) {
             while (pts[i].y == pts[mod(i+1, pts.length)].y) {
                 i += 1;
+                if (i == pts.length - 1) {
+                    return false;
+                }
             }
             return pts[i].y > pts[mod(i+1, pts.length)].y;
         }
         return false;
     };
 
+    // does any point within 5 points of i have a y value greater than (maxima) / less than (minima) (remember y is inverted) EXTREMA_DETECTION_HEIGHT units further than pts[i].y?
+    const isNotable = (i: number, style: TurningPointStyle) => {
+        for (let j = isClosed ? 0 : Math.min(0, i - 5); j < (isClosed ? pts.length : Math.max(pts.length, i + 5)); j++) {
+            if (style === TurningPointStyle.MAXIMA && pts[mod(j, pts.length)].y > pts[i].y + EXTREMA_DETECTION_HEIGHT) {
+                return true;
+            } else if (style === TurningPointStyle.MINIMA && pts[mod(j, pts.length)].y < pts[i].y - EXTREMA_DETECTION_HEIGHT) {
```

Comment:
If `y` is inverted should this be `>` rather than `<` as a `MINIMA` is actually the highest local point? Or has that been rectified at this point?

The inverse being true for the above.